lib/core/directives/search-chip-list.directive.ts

```diff
@@ -0,0 +1,33 @@
+/*!
+ * @license
+ * Copyright 2019 Alfresco Software, Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Directive, ElementRef, Renderer2 } from '@angular/core';
+
+@Directive({
+  // tslint:disable-next-line: directive-selector
+  selector: 'mat-chip-list'
```

Comment:
the problem with this selector is that anyone who will be using mat-chip-list and mat-chip will have these DOM changes even if its not desired. If someone wants just to display in a chip list some visual tags
```html
<mat-chip-list>
  <mat-chip>Tag 1</mat-chip>
  <mat-chip>Tag 2</mat-chip>
  <mat-chip>Tag 3</mat-chip>
</mat-chip-list>
```
will result in 

```html
<mat-chip-list>
  <mat-chip role="button">Tag 1</mat-chip>
  <mat-chip role="button">Tag 2</mat-chip>
  <mat-chip role="button">Tag 3</mat-chip>
</mat-chip-list>
```

Reply:
@pionnegru so is there a way to do this level of selectivity and still make this directive a core directive?

Reply:
since we want a to change the attributes in case of a list of chips that are actionable we can have ```selector: ['adf-a11y-action-chip-list']``` or something close to underline the intent of this directive. Same with `mat-chip`, can be ```selector: ['adf-a11y-action-chip']```.  
Then we can use them like normal directives